src/all/comicgrowl/src/eu/kanade/tachiyomi/extension/all/comicgrowl/ComicGrowl.kt

```diff
@@ -1,63 +1,216 @@
+        val requestUrl = "$baseUrl/book/contentsInfo".toHttpUrl().newBuilder()
+            .addQueryParameter("comici-viewer-id", comiciViewerId)
+            .addQueryParameter("user-id", memberJwt)
+            .addQueryParameter("page-from", "0")
+
+        // Initial request to get total pages
+        val initialRequest = GET(requestUrl.addQueryParameter("page-to", "1").build(), headers)
+        client.newCall(initialRequest).execute().use { initialResponseRaw ->
+            if (!initialResponseRaw.isSuccessful) {
+                throw Exception("Failed to get page list")
+            }
+
+            // Get all pages
+            val pageTo = initialResponseRaw.parseAs<PageResponse>(json).totalPages.toString()
+            val getAllPagesUrl = requestUrl.setQueryParameter("page-to", pageTo).build()
+            val getAllPagesRequest = GET(getAllPagesUrl, headers)
+            client.newCall(getAllPagesRequest).execute().use {
+                if (!it.isSuccessful) {
+                    throw Exception("Failed to get page list")
+                }
+
+                it.parseAs<PageResponse>(json).result.forEach { resultItem ->
+                    // Origin scramble string is something like [6, 9, 14, 15, 8, 3, 4, 12, 1, 5, 0, 7, 13, 2, 11, 10]
+                    val scramble = resultItem.scramble.drop(1).dropLast(1).replace(", ", "-")
+                    // Add fragment to let interceptor descramble the image
+                    val imageUrl = resultItem.imageUrl.toHttpUrl().newBuilder().fragment(scramble).build()
+
+                    pageList.add(
+                        Page(index = resultItem.sort, imageUrl = imageUrl.toString()),
+                    )
+                }
+            }
+        }
+        return pageList
+    }
+
+    override fun imageUrlParse(document: Document): String {
+        throw UnsupportedOperationException()
+    }
 
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
-        if (query.isNotEmpty()) {
-            val url = "$baseUrl/search".toHttpUrl().newBuilder().addQueryParameter("q", query)
+        val searchUrl = "$baseUrl/search".toHttpUrl().newBuilder()
+            .setQueryParameter("keyword", query)
+            .setQueryParameter("page", page.toString())
+            .build()
+        return GET(searchUrl, headers)
+    }
+
+    override fun searchMangaNextPageSelector() = null
+
+    override fun searchMangaSelector() = ".series-list a"
+
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.absUrl("href"))
+        title = element.selectFirst(".manga-title")!!.text()
+        setImageUrlFromElement(element)
+    }
+
+    override fun latestUpdatesRequest(page: Int) = GET(baseUrl, headers)
+
+    override fun latestUpdatesNextPageSelector() = null
+
+    override fun latestUpdatesSelector() = "div.feature-list:nth-child(2) > .feature-item"
```

Comment:
Avoid using offset if possible.
```suggestion
    override fun latestUpdatesSelector() = "h2:contains(新) + .feature-list > .feature-item"
```